src/core/ext/filters/client_channel/lb_policy/xds/xds.cc

```diff
@@ -875,14 +845,6 @@ grpc_channel_args* BuildBalancerChannelArgs(
   };
   // Channel args to add.
   const grpc_arg args_to_add[] = {
-      // New server address list.
-      // Note that we pass these in both when creating the LB channel
-      // and via the fake resolver.  The latter is what actually gets used.
-      CreateServerAddressListChannelArg(balancer_addresses.get()),
-      // The fake resolver response generator, which we use to inject
-      // address updates into the LB channel.
-      grpc_core::FakeResolverResponseGenerator::MakeChannelArg(
```

Comment:
In order to test this, we need a way to inject a fake resolver response generator into the LB channel as well.  I suggest adding a separate xds-specific channel arg to set the resolver response generator for the balancer channel.